### PR TITLE
Spell checking spec

### PR DESCRIPTION
**Pull request by jackfranklin**
This is a PR to tackle issue #1105.

The first commit introduces the spec that ensures each word is used.
The second commit is some words that I found were unused and hence failed the spec.
The third is just some tidying up. I'm happy to squish the commits once this gets into a state of merge-ability.

### CODE REVIEW

File: spec/source_file_finder.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class SourceFileFinder
+  ALLOWED_FILE_EXTENSIONS = [".md"]
```
Comment (jackfranklin) [inline spec/source_file_finder.rb]: There's some SCSS and JS files in `source/` that I presumed we want to ignore - let me know if that assumption is incorrect.

**Comment by jackfranklin** [comment]
It would appear I made a couple of mistakes from the last Travis build, taking a look now :+1:

**Comment by jackfranklin** [comment]
I might need a hand clarifying exactly how the spell check works. For example, on Travis I see:

```
spellcheck  Running spell checker for /object-model/enumerables/
    misspell  The word 'MethodObservable' is misspelled
    misspell  The word 'poppopObject' is misspelled
    misspell  The word 'pushpushObject' is misspelled
    misspell  The word 'reversereverseObjects' is misspelled
    misspell  The word 'shiftshiftObject' is misspelled
    misspell  The word 'unshiftunshiftObject' is misspelled
```

But if I run:

```
cat source/object-model/enumerables.md | grep MethodObservable
```

I don't get anything back. Is this something Middleman is inserting when it generates the build?

**Comment by locks** [comment]
@jackfranklin I think you missed https://github.com/emberjs/guides/pull/1138. FWIW they seem to be having a similar problem

**Comment by jackfranklin** [comment]
@locks ah, fair enough! The implementation looks cleaner, so I'll close in favour of that one.

**Pull request closed without merging**